src/pages/api/multipleOperatorsServiceList.ts

```diff
@@ -0,0 +1,101 @@
+import { NextApiResponse } from 'next';
+import isArray from 'lodash/isArray';
+import { MultipleOperatorsAttribute } from './searchOperators';
+import { MULTIPLE_OPERATOR_ATTRIBUTE, MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE } from '../../constants/index';
+import { isMultiOperatorInfoWithErrors } from '../../interfaces/typeGuards';
+
+import { redirectTo, redirectToError } from './apiUtils';
+import { isSessionValid } from './apiUtils/validator';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
+import { NextApiRequestWithSession, ErrorInfo, MultiOperatorInfo } from '../../interfaces';
+
+const errorId = 'checkbox-0';
+
+export interface MultiOperatorInfoWithErrors {
+    errors: ErrorInfo[];
+}
+
+export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
+    const redirectUrl = '/multipleOperatorsServiceList';
+    const selectAllText = 'Select All Services';
+
+    try {
+        if (!isSessionValid(req, res)) {
+            throw new Error('session is invalid.');
+        }
+
+        const refererUrl = req?.headers?.referer;
+        const queryString = refererUrl?.substring(refererUrl?.indexOf('?') + 1);
+        const { selectAll } = req.body;
+
+        const isSelected = selectAll === selectAllText;
+
+        if (selectAll && queryString) {
+            redirectTo(res, `${redirectUrl}?selectAll=${isSelected}`);
+            return;
+        }
+        const data = getSessionAttribute(req, MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE);
+        let dataToReAdd: MultiOperatorInfo[] = [];
+        if (isMultiOperatorInfoWithErrors(data)) {
+            dataToReAdd = data.multiOperatorInfo;
+        } else if (data) {
+            dataToReAdd = data;
+        }
+        if ((!req.body || Object.keys(req.body).length === 0) && !selectAll) {
+            updateSessionAttribute(req, MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE, {
+                multiOperatorInfo: dataToReAdd,
+                errors: [{ id: errorId, errorMessage: 'Choose at least one service from the options' }],
+            });
+            redirectTo(res, `${redirectUrl}?selectAll=false`);
+            return;
+        }
+
+        updateSessionAttribute(req, MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE, dataToReAdd);
+
+        const selectedServices: string[] = [];
+        const requestBody: { [key: string]: string | string[] } = req.body;
+        let currentNocCode = '';
+
+        Object.entries(requestBody).forEach(entry => {
+            const nocCodeLineNameServiceCodeStartDate = entry[0];
```

Comment:
This is old code (same as service list page) and I cant see what could be more descriptive honestly, I think its named that way because of the format it come through in.